R/cchart.R

```diff
@@ -269,19 +269,273 @@ CChart <- function(chart.type, x, small.multiples = FALSE,
+        series.settings <- list(list(
+            CustomPoints = getColorsAsNumericScale(data, args$colors, tmp.opacity),
+            Marker = list(Size = args$marker.size, OutlineStyle = "None"),
+            ShowDataLabels = tmp.data.label.show,
+            DataLabelsFont = list(family = args$data.label.font.family,
+                size = args$data.label.font.size/1.3333,
+                color = tmp.data.label.font.color[1]),
+            OutlineStyle = "None"))
+
+    } else if (chart.type %in% c("BarMultiColor", "ColumnMultiColor",
+               "Pyramid", "Bar Pictograph"))
+    {
+        # Multi-color series is implemented as a single series
+        # with many CustomPoints
+        tmp.colors <- list()
+        for (i in 1:length(args$colors))
+            tmp.colors[[i]] <- list(BackgroundColor = sprintf("%s%X",
+                args$colors[i], round(tmp.opacity*255)), Index = i - 1)
+        series.settings <- list(list(
+            CustomPoints = tmp.colors,
+            ShowDataLabels = tmp.data.label.show,
+            DataLabelsFont = list(family = args$data.label.font.family,
+                size = args$data.label.font.size/1.3333,
+                color = tmp.data.label.font.color[1]),
+            DataLabelsPosition = tmp.data.label.position,
+            OutlineColor = tmp.line.color[1], # style is none if no border color defined
+            OutlineWidth = tmp.line.thickness[1],
+            OutlineStyle = tmp.line.style))
+
+    } else
+        series.settings <- lapply(1:length(args$colors),
+        function(i) {list(
+            BackgroundColor = sprintf("%s%X", args$colors[i], round(tmp.opacity*255)),
+            ShowDataLabels = tmp.data.label.show,
+            ShowCategoryNames = tmp.data.label.show.category.labels,
+            DataLabelsFont = list(family = args$data.label.font.family,
+                size = args$data.label.font.size/1.3333,
+                color = tmp.data.label.font.color[i]),
+            DataLabelsPosition = tmp.data.label.position,
+            OutlineColor = tmp.line.color[i],
+            OutlineWidth = tmp.line.thickness[i],
+            OutlineStyle = tmp.line.style)})
+    tmp.n <- length(series.settings)
+
+
+    if ((chart.type == "Scatter" && isTRUE(args$scatter.colors.as.categorical)) || chart.type == "Line")
+        for (i in 1:tmp.n)
+            series.settings[[i]]$Marker = list(Size = args$marker.size,
+                OutlineStyle = "None",
+                BackgroundColor = sprintf("%s%X", args$colors[i], round(tmp.opacity*255)))
+
+    # Initialise return output
+    res <- list()
+    if (length(series.settings) > 0)
+        res$TemplateSeries = series.settings
+    if (isTRUE(args$legend.show == FALSE) || isTRUE(args$legend.show == "Hide"))
+        res$ShowLegend <- FALSE
+    res$Legend = list(Font = list(color = args$legend.font.color,
+            family = args$legend.font.family, size = args$legend.font.size/1.3333))
```

Comment:
Why divide by 1.3333? Perhaps it should be a variable with a descriptive name?

Reply:
Or a function with a descriptive name that divides by 1.3333

Reply:
Replaced the 1.333 by a function called `px2pt` in a060c6a